mono/utils/mono-error.c

```diff
@@ -611,6 +623,10 @@ mono_error_prepare_exception (MonoError *oerror, MonoError *error_out)
 		exception = mono_get_exception_argument (error->type_name, mono_internal_error_get_message (error));
 		break;
 
+	case MONO_ERROR_ARGUMENT_NULL:
```

Comment:
Unfortunately, new error kinds require us to support the conversion to loader error for now.

Meaning you need to change mono_error_set_from_loader_error and mono_loader_set_error_from_mono_error too.

Ideally this would be a separate commit from the one that introduced MonoError into mono_object_register_finalizer.